packages/dapp-high-roller/src/components/app-game/app-game.tsx

```diff
@@ -174,61 +103,73 @@ export class AppGame {
   }
 
   async handleRoll(): Promise<void> {
-    if (this.isProposing) {
-      if (this.highRollerState.stage === HighRollerStage.PRE_GAME) {
-        await Promise.all([
-          this.animateRoll("myRoll"),
-          this.animateRoll("opponentRoll")
-        ]);
-        const startGameAction: Action = {
-          number: 0,
-          actionType: ActionType.START_GAME,
-          actionHash: HashZero
-        };
-
-        this.highRollerState = await takeAction(startGameAction); // TODO call appInstance.takeAction
-
-        const numberSalt =
-          "0xdfdaa4d168f0be935a1e1d12b555995bc5ea67bd33fce1bc5be0a1e0a381fc90";
-        const playerFirstNumber = Math.floor(Math.random() * Math.floor(1000));
-        const hash = computeCommitHash(numberSalt, playerFirstNumber);
-
-        const commitHashAction: Action = {
-          number: playerFirstNumber,
-          actionType: ActionType.COMMIT_TO_HASH,
-          actionHash: hash
-        };
-
-        this.highRollerState = await takeAction(commitHashAction); // TODO call appInstance.takeAction
-
-        onUpdateState().then((state: HighRollerAppState) => {
-          // TODO use updateState event in app-provider
-          const rolls = this.highRoller(
-            state.playerFirstNumber,
-            state.playerSecondNumber
-          );
-          this.myRoll = rolls.myRoll;
-          this.opponentRoll = rolls.opponentRoll;
-          const totalMyRoll = this.myRoll[0] + this.myRoll[1];
-          const totalOpponentRoll = this.opponentRoll[0] + this.opponentRoll[1];
-          if (totalMyRoll > totalOpponentRoll) {
-            this.myScore += 1;
-            this.gameState = GameState.Won;
-          } else if (totalMyRoll < totalOpponentRoll) {
-            this.opponentScore += 1;
-            this.gameState = GameState.Lost;
-          } else {
-            this.gameState = GameState.Tie;
-          }
-          this.highRollerState = state;
-        });
-      }
+    if (!this.appInstance.takeAction) {
```

Comment:
Since this doesn't differentiate between `isProposing` does it mean that the accepting player can be `playerFirst`?